src/system-program.js

```diff
@@ -139,13 +144,27 @@ const SystemInstructionLayout = Object.freeze({
       BufferLayout.ns64('amount'),
     ]),
   },
+  CreateWithSeed: {
+    index: 3,
+    layout: BufferLayout.struct([
+      BufferLayout.u32('instruction'),
+      Layout.rustString('seed'),
+      BufferLayout.ns64('lamports'),
+      BufferLayout.ns64('space'),
+      Layout.publicKey('programId'),
+    ]),
+  },
 });
 
 /**
  * Populate a buffer of instruction data using the SystemInstructionType
  */
 function encodeData(type: SystemInstructionType, fields: Object): Buffer {
-  const data = Buffer.alloc(type.layout.span);
+  const allocLength =
```

Comment:
more magic?

Reply:
Yes. I'm not entirely sure this is the best way to handle the the rust string, but it seemed reasonably extensible for other value-specific layouts, and I didn't see a more straightforward way using the `buffer-layout` package.

Reply:
Can we just use a simple `char *` string?  Requiring utf8 is unnecessarily restrictive and is annoying to handle -- ie, all this code

Reply:
I'm not sure I see what you're envisioning, @mvines . The gymnastics in this pr are largely around handling dynamically sized layouts in buffer-layout, which would still be an issue even if not utf8, right?

Reply:
I guess it seems wasteful to require a length of 8 bytes for a seed with a length of 32 bytes or less

Reply:
Ah, yeah. I expect we could use shortvec easily enough here, if @rob-solana likes that option upstream.

Reply:
I've no preferences for how this side comes out...

for the system instruction, my goals are:
1. looks pretty for rust clients and for the implementation
2. supports a reasonable-sized mnemonic (32 is a nice length if one stays in ascii)
3. limit the space the seed can cover to something smaller than 256-bits (ascii and utf-8 have those restrictions)